gaphor/UML/tests/test_umllex.py

```diff
@@ -145,6 +145,22 @@ def test_parse_association_end_derived_end(factory):
     assert not p.defaultValue
 
 
+def test_parse_association_end_with_type(factory):
+    """Test parsing of association end, type is ignored
+    """
+    a = factory.create(UML.Association)
+    p = factory.create(UML.Property)
+    p.association = a
+
+    UML.parse(p, "end: TypeVal")
+    assert "end" == p.name
+    assert not p.typeValue
+    assert not p.lowerValue
+    assert not p.upperValue
+    assert not p.defaultValue
+    assert not p.typeValue
```

Comment:
Is there supposed to be two assert statements for `p.typeValue`?

Reply:
Fixed